### DISCUSSION TITLE
Get data from async functions

### DISCUSSION THREAD

**Discussion by ghost** [alpinejs/alpine#1047] [3. Feature ideas]
It would be nice if we could get data from async functions (awaiting for it, off course).

```html
    <label x-text="fn()"></label>

    <script>
      async function fn(key) {
        return 'value';
      }
    </script>
```

We had some discussion about it here:
https://github.com/alpinejs/alpine/discussions/1042

**Comment by ryangjchandler** [comment]
@fabianobonin Feel free to open a pull request with the necessary changes. Opening a new discussion won't make it happen any quicker 😉

**Reply by HugoDF** [reply]
I think it's because by default feature requests get routed to a new discussion

You can create an issue using this link https://github.com/alpinejs/alpine/issues/new

**Reply by ghost** [reply]
I am not a experienced javascript developer at all...
Would be very glad to implement it myself, but i don't think i am able to...
Anyway, i will give it a try, but will probably write some code that will make me feel ashamed later 😅

Best regards

**Reply by ghost** [reply]
Sorry, I didn't understand your message at first.

I tried to submit a new issue, but was redirected to a new discussion.

It's done in the right place now.

**Reply by ghost** [reply]
Please delete this topic if you can, I was unable to do it myself.

Thanks
